packages/xo-server/src/xapi/mixins/pool.mjs

```diff
@@ -87,90 +86,107 @@ const methods = {
     }
     ;[hosts[0], hosts[indexOfMaster]] = [hosts[indexOfMaster], hosts[0]]
 
-    let hasRestartedOne = false
     // Restart all the hosts one by one
-    for (const host of hosts) {
-      const hostId = host.uuid
-      if (ignoreHost && ignoreHost(host)) {
-        continue
-      }
-
-      // This is an old metrics reference from before the pool master restart.
-      // The references don't seem to change but it's not guaranteed.
-      const metricsRef = host.metrics
-
-      await this.barrier(metricsRef)
-      await this._waitObjectState(metricsRef, metrics => metrics.live)
-
-      const getServerTime = async () => parseDateTime(await this.call('host.get_servertime', host.$ref)) * 1e3
-      log.debug(`Evacuate host ${hostId}`)
-      await this.clearHost(host)
-
-      if (beforeRebootHost) {
-        await beforeRebootHost(host)
+    await Task.run(
+      { properties: { name: `Restarting hosts`, total: hosts.length, progress: 0, done: 0 } },
+      async () => {
+        let done = 0
+        for (const host of hosts) {
+          const hostId = host.uuid
+          const hostName = host.name_label
+
+          if (!ignoreHost || !ignoreHost(host)) {
+            await Task.run({ properties: { name: `Restarting host ${hostId}`, hostId, hostName } }, async () => {
+              // This is an old metrics reference from before the pool master restart.
+              // The references don't seem to change but it's not guaranteed.
+              const metricsRef = host.metrics
+
+              await this.barrier(metricsRef)
+              await this._waitObjectState(metricsRef, metrics => metrics.live)
+
+              const getServerTime = async () => parseDateTime(await this.call('host.get_servertime', host.$ref)) * 1e3
+              await Task.run({ properties: { name: `Evacuate`, hostId, hostName } }, async () => {
+                await this.clearHost(host)
+              })
+
+              if (beforeRebootHost) {
+                await beforeRebootHost(host)
+              }
+
+              const rebootTime = await getServerTime()
+              await Task.run({ properties: { name: `Restart`, hostId, hostName } }, async () => {
+                await this.callAsync('host.reboot', host.$ref)
+              })
+
+              await Task.run({ properties: { name: `Waiting for host to be up`, hostId, hostName } }, async () => {
+                await timeout.call(
+                  (async () => {
+                    await this._waitObjectState(
+                      hostId,
+                      host => host.enabled && rebootTime < host.other_config.agent_start_time * 1e3
+                    )
+                    await this._waitObjectState(metricsRef, metrics => metrics.live)
+                  })(),
+                  this._restartHostTimeout,
+                  new Error(`Host ${hostId} took too long to restart`)
+                )
+              })
+              Task.info(`Host ${hostId} is up`)
```

Comment:
Does this provide more information that the task above?

If not, remove.

Reply:
It is indeed redundant with the task status